src/KubeOps/Operator/Events/EventManager.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using DotnetKubernetesClient;
+using k8s;
+using k8s.Models;
+using KubeOps.Operator.Entities.Extensions;
+using Microsoft.Extensions.Logging;
+using SimpleBase;
+
+namespace KubeOps.Operator.Events
+{
+    internal class EventManager : IEventManager
+    {
+        private readonly IKubernetesClient _client;
+        private readonly OperatorSettings _settings;
+        private readonly ILogger<EventManager> _logger;
+
+        public EventManager(IKubernetesClient client, OperatorSettings settings, ILogger<EventManager> logger)
+        {
+            _client = client;
+            _settings = settings;
+            _logger = logger;
+        }
+
+        public async Task Publish(
+            IKubernetesObject<V1ObjectMeta> resource,
+            string reason,
+            string message,
+            EventType type = EventType.Normal)
+        {
+            _logger.LogTrace(
+                "Encoding event name with: {resourceName}.{resourceNamespace}.{reason}.{message}.{type}.",
+                resource.Name(),
+                resource.Namespace(),
+                reason,
+                message,
+                type);
+            var eventName =
+                Base32.Rfc4648.Encode(
+                    SHA512.HashData(
+                        Encoding.UTF8.GetBytes($"{resource.Name()}.{resource.Namespace()}.{reason}.{message}.{type}")));
+            _logger.LogTrace(@"Search or create event with name ""{name}"".", eventName);
+            var @event = await _client.Get<Corev1Event>(eventName, resource.Namespace()) ??
+                         new Corev1Event
```

Comment:
Does it only support core events?

Reply:
By now, yes. I found out that the events.k8s.io/v1 and events.k8s.io/v1beta1 are only on v1.19 onward. So they are practically nowhere.

The CoreV1Events are the ones that are shown on `describe` and `kubectl get events` commands.

Reply:
Also, some kind of "version gate" could be a nice improvement.
So one could mask the effective version of the event and determine based on the kubernetes server veresion, which implementation should be used.